### PR TITLE
feat: app state migrations

### PR DESCRIPTION
**Pull request by pvinis**
This PR resolves [MOPLAT-478]

### Description

moving migrations from eigen to energy. i tried with ezpz persist (doesn't include migrations), redux-persist (doesn't work well with ezpz's computed values), so i took the working-very-well @ds300's work from eigen.

### PR Checklist

- [ ] I tested my changes on **iOS** and **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[docs]: /docs/README.md

[MOPLAT-478]: https://artsyproduct.atlassian.net/browse/MOPLAT-478?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/screens/Albums/AlbumTabs/AlbumArtworks.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ import { extractNodes } from "app/utils/extractNodes"
 import { usePresentationFilteredArtworks } from "app/utils/hooks/usePresentationFilteredArtworks"
 
 export const AlbumArtworks = ({ artworkIds }: { artworkIds: string[] }) => {
-  const partnerID = GlobalStore.useAppState((state) => state.activePartnerID)!
+  const partnerID = GlobalStore.useAppState((state) => state.auth.activePartnerID)!
```
Comment (damassi) [inline src/app/screens/Albums/AlbumTabs/AlbumArtworks.tsx:15]: 👍

File: src/app/system/store/GlobalStore.tsx [hunk 2]
```diff
@@ -77,10 +46,13 @@ export const __globalStoreTestUtils__ = __TEST__
...
       getCurrentState: () => globalStoreInstance.getState(),
       dispatchedActions: [] as Action[],
+      reset: () => {
+        globalStoreInstance = createGlobalStore()
+      },
     }
   : undefined
 
-const globalStoreInstance = createGlobalStore()
+let globalStoreInstance = createGlobalStore()
```
Comment (damassi) [inline src/app/system/store/GlobalStore.tsx:55]: Can we move this up to line 28?

File: src/app/system/store/Models/AuthModel.ts [hunk 3]
```diff
@@ -16,18 +16,21 @@ interface AuthModelState {
...
 }
 
 const authModelInitialState: AuthModelState = {
   userAccessToken: null,
   userAccessTokenExpiresIn: null,
   userID: null,
   xAppToken: null,
-  xApptokenExpiresIn: null,
+  xAppTokenExpiresIn: null,
+  activePartnerID: null,
```
Comment (damassi) [inline src/app/system/store/Models/AuthModel.ts:29]: ditto

File: src/app/system/store/Models/DevicePrefsModel.ts [hunk 4]
```diff
@@ -54,6 +57,11 @@ export const getDevicePrefsModel = (): DevicePrefsModel => ({
     state.showDevMenuButtonInternalToggle = option
   }),
 
+  offlineSyncedChecksum: "never-happened",
```
Comment (damassi) [inline src/app/system/store/Models/DevicePrefsModel.ts:60]: This reads a bit funny in the app offline page... can we change to "None"?

**Review by damassi — APPROVED** [review]
This looks good! A couple of superminor comments.

I still really wish we could lean on OSS here and figure out how to make it work vs adding more hand rolled David code, but its what we do in Eigen and folks know it and it works. At some point I would like to take a stab at it though.

**Comment by pvinis** [comment]
converting to draft because i see some funky things on login.

File: src/app/system/store/Models/DevicePrefsModel.ts (hunk 4 above)
Comment (pvinis) [reply src/app/system/store/Models/DevicePrefsModel.ts:60]: how about `N/A`?

**Pull request closed without merging**